argus/scenarios/cloud.py

```diff
@@ -144,33 +159,42 @@ def get_network_interfaces(self):
         guest_nics = []
         for network in self._networks or []:
             network_id = network["uuid"]
-            details = net_client.show_network(network_id)["network"]
-            subnet_id = details["subnets"][0]
-            details = net_client.show_subnet(subnet_id)["subnet"]
-
-            # The network interface should follow the format found under
-            # `windows.InstanceIntrospection.get_network_interfaces` method.
-            nic = {}
-            nic["dhcp"] = details["enable_dhcp"]
-            nic["dns"] = details["dns_nameservers"]
-            nic["gateway"] = details["gateway_ip"]
-            nic["netmask"] = util.cidr2netmask(details["cidr"])
-
-            # Find rest of the details under the ports using this subnet.
-            # There should be no conflicts because on the current architecture
-            # every instance is using its own router, subnet and network
-            # accessible only to it.
-            ports = net_client.list_ports()["ports"]
-            for port in ports:
-                # Select instance related ports only, with the
-                # corresponding subnet ID.
-                if ("compute" not in port["device_owner"] or
-                        port["fixed_ips"][0]["subnet_id"] != subnet_id):
-                    continue
-
-                nic["mac"] = port["mac_address"].upper()
-                nic["address"] = port["fixed_ips"][0]["ip_address"]
-                break
+            net_details = net_client.show_network(network_id)["network"]
+            nic = dict.fromkeys(util.NETWORK_KEYS)
+            for subnet_id in net_details["subnets"]:
+                details = net_client.show_subnet(subnet_id)["subnet"]
+
+                # The network interface should follow the format found under
+                # `windows.InstanceIntrospection.get_network_interfaces`
+                # method or `argus.util.NETWORK_KEYS` model.
+                v6switch = details["ip_version"] == 6
+                v6suffix = lambda: "6" if v6switch else ""
+                nic["dhcp"] = details["enable_dhcp"]
+                nic["dns" + v6suffix()] = details["dns_nameservers"]
+                nic["gateway" + v6suffix()] = details["gateway_ip"]

```

Comment:
Yes, they are on the same name, but they contain different type values depending on `ip_version`.